### PR TITLE
[Feature] Enable AMX FP16 on next generation CPU

### PR DESCRIPTION
**Pull request by wenhuanh**
need export env DNNL_MAX_CPU_ISA=AVX512_CORE_AMX_FP16 when running

### CODE REVIEW

File: src/utils/matmul_helper.h [hunk 1]
```diff
@@ -1829,11 +1852,11 @@ class MMHelper {
         stream->wait();
     }
 
-    template <typename Tin, typename Tout>
-    void onednn_amx_sgemm_f32bf16f32_compute(bool transA, int M, int N, int K, float alpha, const Tin *A, int lda,
-            const bfloat16_t *packedB, float beta, Tout *C, int ldc, const matmul_kinds postAlg = matmul_kinds::Basic) {
-        TimeLine t("onednn_amx_sgemm_f32bf16f32_compute");
-        TimeLine t1("onednn_amx_sgemm_f32bf16f32_compute.create_primitive");
+    template <typename Tin, typename Tweight, typename Tout>
+    void onednn_amx_sgemm_f32x16f32_compute(bool transA, int M, int N, int K, float alpha, const Tin *A, int lda,
```
Comment (changqi1) [inline src/utils/matmul_helper.h:1856]: reanme onednn_amx_gemm_compute?

File: src/utils/matmul_helper.h [hunk 2]
```diff
@@ -2213,12 +2248,12 @@ class MMHelper {
         stream->wait();
     }
 
-    template <typename Tin, typename Tout>
-    void onednn_amx_sgemm_f32bf16f32_compute_residential(bool transA, int M, int N, int K, float alpha, const Tin *A,
-            int lda, const bfloat16_t *packedB, float beta, Tout *C, int ldc, const float *bias, const Tin *res,
+    template <typename Tin,  typename Tweight, typename Tout>
+    void onednn_amx_sgemm_f32x16f32_compute_residential(bool transA, int M, int N, int K, float alpha, const Tin *A,
```
Comment (changqi1) [inline src/utils/matmul_helper.h:2252]: reanme onednn_amx_gemm_compute_residential?

**Comment by changqi1** [comment]
除了compute和compute_resident，其他的kernel也需要支持这个amx fp16的支持

**Comment by wenhuanh** [comment]
revised and updated.

**Comment by changqi1** [comment]
We will need to check ISA to select AMX_FP16 or AVX512_FP16. But currently this is OK.

**Comment by wenhuanh** [comment]
Perf status was updated in related JIRA.

**Review by changqi1 — APPROVED** [review]

**Pull request merged**